cpp/src/slice2cpp/CPlusPlusUtil.cpp

```diff
@@ -502,7 +502,7 @@ Slice::typeToString(
     ClassDeclPtr cl = dynamic_pointer_cast<ClassDecl>(type);
     if (cl)
     {
-        return getUnqualified(cl->scoped(), scope) + "Ptr";
+        return getUnqualified(fixKwd(cl->scoped() + "Ptr"), scope);
```

Comment:
I suspect the original writer thought that they didn't need to escape because we put `Ptr` on the end,
which makes it impossible to hit a keyword. _However_ `fixKeyword` also fixes the names of the _scopes_...
which is what was failing here.

Also, it's more correct to add any suffixes _before_ we escape.

Reply:
This escaping is not correct. As documented:
https://doc.zeroc.com/ice/3.7/language-mappings/c++11-mapping/client-side-slice-to-c++11-mapping/c++11-mapping-for-identifiers

we escape only when there is no suffix. It would be whilePtr, not _cpp_whilePtr.

Reply:
My comment is incorrect - whilePtr would not get escaped by the call above.

Reply:
Yes, `fixKwd` will only 'fix' the passed identifier if necessary. Since we add the `Ptr` before we call it,
the identifier will never be escaped, as per the docs.

But, calling this function is still necessary to fix the _scopes_ of the identifier:
`::and::while` needs to become: `::_cpp_and::whilePtr`